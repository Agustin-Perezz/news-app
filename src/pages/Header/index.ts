export * from './Components/NavBar';
export * from './Components/SearchBar';
export { default as HeaderPage } from './HeaderPage';
