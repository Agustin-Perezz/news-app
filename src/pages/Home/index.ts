export { default as HomePage } from './components/HomePage';
export * from './components';
