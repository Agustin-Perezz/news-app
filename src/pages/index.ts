export * from './Categories';
export * from './Header';
export * from './Home';
export * from './News';
