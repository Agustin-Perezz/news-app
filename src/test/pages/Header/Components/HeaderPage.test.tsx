import { screen } from '@testing-library/react';
import { HeaderPage } from '../../../../pages';
import { renderWhitContext } from '../../../utils';

describe('Test in <HeaderPage />', () => {
  function setup() {
    return renderWhitContext({ children: <HeaderPage />, pathname: '/search', cache: {} });
  }

  it('should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('should render main content', () => {
    setup();
    screen.getByAltText('news-logo');
    screen.getByPlaceholderText('Search any news..');
    screen.getByTestId('GitHubIcon');
    screen.getByTestId('TwitterIcon');
    screen.getByTestId('LinkedInIcon');
  });
});
