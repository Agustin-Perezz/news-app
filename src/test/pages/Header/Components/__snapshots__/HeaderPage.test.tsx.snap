// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <HeaderPage /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1e2ky5m"
  >
    <div
      class="MuiBox-root css-agauc4"
    >
      <li
        class="MuiImageListItem-root css-17u9esh-MuiImageListItem-root"
        style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
      >
        <img
          alt="news-logo"
          class="MuiImageListItem-img"
          src="[object Object]"
        />
      </li>
      <form
        class="MuiBox-root css-p2ap1"
      >
        <div
          class="css-jmprdh"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-jplnuw-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-vvrtyw-MuiInputBase-root"
          >
            <input
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              inputmode="search"
              placeholder="Search any news.."
              type="input"
              value=""
            />
          </div>
        </div>
      </form>
      <div
        class="css-6mk4pn-MuiStack-root"
      >
        <a
          href="https://github.com/Agustin-Perezz"
          target="_black"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-4blva6-MuiAvatar-root"
            icon="[object Object]"
            url="https://github.com/Agustin-Perezz"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
          </div>
        </a>
        <a
          href="https://www.linkedin.com/in/agustinperez-front-end-developer/"
          target="_black"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-4blva6-MuiAvatar-root"
            icon="[object Object]"
            url="https://www.linkedin.com/in/agustinperez-front-end-developer/"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LinkedInIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
          </div>
        </a>
        <a
          href="https://twitter.com/agustinperez__"
          target="_black"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-4blva6-MuiAvatar-root"
            icon="[object Object]"
            url="https://twitter.com/agustinperez__"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="TwitterIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiBox-root css-mhl23k"
      >
        <div
          class="MuiTabs-root css-1ujnqem-MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-bottom: 0px;"
          >
            <div
              aria-label="category-tabs"
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="0"
                type="button"
              >
                default-tab
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                home
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                general
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                science
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                sports
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                business
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                health
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                entertainment
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                tech
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                politics
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                food
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                travel
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-xyhd53-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
